docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
+
+## Current Challenge
+The lack of clear criteria for classifying components as core or non-core has led to:
+- Inconsistent decision-making across different features and plugins
+- Potential bloat in core components
+- Uncertainty in plugin development and maintenance strategies
+
+## Proposed Criteria for Core Components
+
+A component should be considered "core" if it meets the following criteria:
```

Comment:
> if it meets the following criteria

meets all bullet points in all 3 criteria or any one would suffice?

Reply:
I am thinking we will have a score system, and each bullet point will have its score. If the sum of the points reaches the bar we aligned, the component should be put inside core.

Reply:
Will we have to evaluate these for every existing plugin in OSD core?